tests/mcp/features/step_definitions_test.go

```
@@ -0,0 +1,479 @@
+		return fmt.Errorf("tool call failed: %w", tc.toolCallErr)
+	}
+	if tc.toolCallResult != nil && tc.toolCallResult.IsError {
+		text := extractToolResultText(tc.toolCallResult)
+		return fmt.Errorf("tool call returned error: %s", text)
+	}
+	return nil
+}
+
+func (tc *mcpTestContext) theToolCallShouldReturnAnError() error {
+	if tc.toolCallErr != nil {
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 298) but it returns nil

(nilerr)